tests/test_server_different_port.py

```diff
@@ -0,0 +1,48 @@
+import requests
+import multiprocessing as mp
+
+# NOTE: HAD TO ADD THIS TO RUN LOCALLY W/O ERROR
+if __name__ == "__main__":
+    mp.set_start_method("fork")  # Try using 'spawn' if 'fork' doesn't work
+
```

Comment:
it should work without manually setting the start method. May I know what was the error, please?

Reply:
This is the error received without this snippet:

```
File '/Users/grantgasser/Documents/Development/LitServe/src/litserve/python_client.py' copied to '/Users/grantgasser/Documents/Development/LitServe/client.py'
Port 8080 has been set in 'dest_path'
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
ERROR:    [Errno 48] Address already in use
Traceback (most recent call last):
  File "/Users/grantgasser/Documents/Development/LitServe/tests/test_server_different_port.py", line 46, in <module>
    test_server_different_port()
  File "/Users/grantgasser/Documents/Development/LitServe/tests/test_server_different_port.py", line 38, in test_server_different_port
    server.run(port=non_default_port_no)
  File "/Users/grantgasser/Documents/Development/LitServe/src/litserve/server.py", line 459, in run
    manager, litserve_workers = self.launch_inference_worker(num_api_servers)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/grantgasser/Documents/Development/LitServe/src/litserve/server.py", line 210, in launch_inference_worker
    manager = mp.Manager()
              ^^^^^^^^^^^^
  File "/Users/grantgasser/miniconda3/envs/litserve-env/lib/python3.11/multiprocessing/context.py", line 57, in Manager
    m.start()
  File "/Users/grantgasser/miniconda3/envs/litserve-env/lib/python3.11/multiprocessing/managers.py", line 567, in start
    self._address = reader.recv()
                    ^^^^^^^^^^^^^
  File "/Users/grantgasser/miniconda3/envs/litserve-env/lib/python3.11/multiprocessing/connection.py", line 250, in recv
    buf = self._recv_bytes()
          ^^^^^^^^^^^^^^^^^^
  File "/Users/grantgasser/miniconda3/envs/litserve-env/lib/python3.11/multiprocessing/connection.py", line 430, in _recv_bytes
    buf = self._recv(4)
          ^^^^^^^^^^^^^
  File "/Users/grantgasser/miniconda3/envs/litserve-env/lib/python3.11/multiprocessing/connection.py", line 399, in _recv
    raise EOFError
EOFError
```

Reply:
it seems like something is already running in port 8080. could you please check that `lsof -i :8080`?

Reply:
correct, killed those processes and things seem to be fine now :+1: